Change NOT to ~

yoni206 opened the pull request:
On my machine `NOT` didn't work, but `~` did.

hrkrshnn commented:
The CI job is failing. Does `scripts/run_proofs.sh` run without problems in your system?

leonardoalt commented:
@yoni206 could this be a python-z3 version thing? The script runs without problems on my machine

leonardoalt commented:
Ok I think the `NOT` is actually correct, where it's trying to use the opcode `NOT` which should be in `opcodes.py` but is not.
@yoni206 the effect is going to be the same, but could you please add `NOT` to `opcodes.py` (after `AND` and `OR`) which returns `~`?

leonardoalt commented:
The CI problem is that the script runs `python3` but apparently `z3-solver` is only installed for `python2.7`.

leonardoalt commented:
The CI problem should go away now that https://github.com/ethereum/solidity/pull/8709 was merged and this can be rebased

yoni206 commented:
@leonardoalt I added `NOT` and rebased locally. Locally my diff to ethereum/solidity develop branch is only the addition of `NOT` but here there are more changes for some reason.

leonardoalt commented:
@yoni206 the last commit looks like a merge of your feature branch. Did you do a merge at some point?

yoni206 commented:
yes, wanted to be up to date with develop. no good?

leonardoalt commented:
Yep, but we usually only `rebase` on top of `develop` for that.

leonardoalt commented:
So that the history of the feature branch is clean, and the only merge commit is from the PR.

yoni206 committed:
Adding bit-vector NOT operation to the opcodes.

yoni206 commented:
now better?

leonardoalt commented:
Yes :) thanks
Sorry for being pedantic but just one more thing since we're at it.. Can you change the commit message to something a little more expressive?

yoni206 commented:
No problem, please let me know if a better message is possible :)

leonardoalt reviewed (approved):
That's good, thanks!
